combat-trainer.lic

```diff
@@ -1457,18 +1457,16 @@ class SpellProcess
     game_state.casting_weapon_buff = $weapon_buffs.include?(name)
     if data['ritual']
       cast_ritual(data, game_state)
-    elsif data['runestone_exists']
-      #DRC.message("runestone_exists setting is set")
-      if data['runestone_for_combat_exists']==false
-        #DRC.message("In if runestone for combat exists logic and it is: #{data['runestone_for_combat_exists']}")
-        @buff_spells.reject! { |spell| @buff_spells[spell]['runestone_for_combat_exists']==false }
-        DRC.message("name array after reject: #{@buff_spells}")
-      end
-      if DRCI.inside?(data['runestone'], @settings.runestone_storage)
+    elsif data['runestone_name']
+      if DRCI.inside?(data['runestone_name'], @settings.runestone_storage)
         data['runestone_for_combat_exists']=true
+        #echo(@buff_spells)
       else
+        #echo(@buff_spells)
+        ### Just need this setting to remove the spell from the list
         data['runestone_for_combat_exists']=false
-        #DRC.message(@buff_spells)
+        @buff_spells.reject! { |spell| @buff_spells[spell]['runestone_for_combat_exists']==false }
+        #echo(@buff_spells)
         return
       end
       cast_runestone(data, game_state) if data['runestone_for_combat_exists']=true
```

Comment:
💡 Since `cast_runestone` is only called when the rune for combat exists, and that boolean is being set to true or false in the if/else at lines 1462 and 1467, I think lines 1460-1476 could be consolidated to this:

```ruby
def check_buffs(game_state)
    ...
    elsif data['runestone_name']
      if DRCI.inside?(data['runestone_name'], @settings.runestone_storage)
        data['runestone_for_combat_exists'] = true
        cast_runestone(data, game_state)
      else
        # Just need this setting to remove the spell from the list
        data['runestone_for_combat_exists'] = false
        @buff_spells.reject! { |spell| @buff_spells[spell]['runestone_for_combat_exists'] == false }
      end
  else
    prepare_spell(data, game_state, @buff_force_cambrinth)
  end
end
```